docs/Migration_Guide_Zephyr.md

```diff
@@ -0,0 +1,587 @@
+# Migrating your code from the Golioth Zephyr SDK to the Golioth Firmware SDK
+
+Golioth support for Zephyr RTOS is included in the [Golioth Firmware
+SDK](https://github.com/golioth/golioth-firmware-sdk). All new projects,
+and updates to existing projects, should use this SDK.
+
+All the great Zephyr support you know and love is now packaged in one
+Golioth repository to rule them all. From the API standpoint, the change
+is mostly naming and syntax. We'll walk you through how to make those
+updates in your code. Under the hood, this change means we can develop,
+test, and rollout new features and better performance to all of our
+supported platforms so you always have access to the best Golioth has to
+offer.
+
+The [Golioth Zephyr SDK](https://github.com/golioth/golioth-zephyr-sdk)
+will be deprecated later in 2024. This migration guide details the
+actions necessary to an existing Zephyr project to the Golioth Firmware
+SDK.
+
+#### Additional resources to help migration:
+- [Golioth Firmware SDK Doxygen
+  reference](https://firmware-sdk-docs.golioth.io/)
+- [Golioth Firmware SDK example
+  code](https://github.com/golioth/golioth-firmware-sdk/tree/main/examples/zephyr)
+- [Golioth Docs: Golioth Firmware
+  SDK](https://docs.golioth.io/firmware/golioth-firmware-sdk/)
+- [Contact the Golioth Developer Relations
+  team](mailto:devrel@golioth.io) with migration questions.
+
+## Migration Overview
+
+1. [Update `west` manifest](#update-west-manifest)
+2. [Update Kconfig files](#update-kconfig-files)
+3. [Update header file includes](#update-header-file-includes)
+4. [Change Golioth API calls](#change-golioth-api-calls)
+5. [Special case: Changes for OTA Firmware
```

Comment:
These special case links don't seem to work in the "rich diff" preview in Github